pkg/controllers/provisioning/provisioner.go

```diff
@@ -368,6 +374,23 @@ func (p *Provisioner) Create(ctx context.Context, n *scheduler.NodeClaim, opts .
 	return nodeClaim.Name, nil
 }
 
+func (p *Provisioner) globalNodeClaimLimitsExceeded(ctx context.Context) error {
+	nodeClaimLimit := options.FromContext(ctx).MaxNodeClaims
+	if nodeClaimLimit < 0 {
+		return nil
+	}
+
+	nodeClaims := &v1beta1.NodeClaimList{}
+	if err := p.kubeClient.List(ctx, nodeClaims); err != nil {
+		return fmt.Errorf("failed to list NodeClaims: %w", err)
+	}
+	if nodeClaimLimit <= len(nodeClaims.Items) {
+		return fmt.Errorf("number of NodeClaims has reached or exceeded global limit of %v", nodeClaimLimit)
+	}
+
+	return nil
+}
+
```

Comment:
Could we use cluster state instead of another list call here? We should have the number of nodeclaims in state nodes.